source/renderer/app/api/utils/apiHelpers.ts

```diff
@@ -27,22 +27,23 @@ export const doesWalletRequireAdaToRemainToSupportTokens = (
   hasAssetsRemainingAfterTransaction?: boolean
 ): {
   requiresAdaToRemainToSupportNativeTokens: boolean;
-  adaToRemain?: number;
+  adaToProceed?: number;
 } => {
   const adaToProceedRegex = new RegExp(
-    /.*I need approximately([\s\d.,]+)ada to proceed.*/
+    /I need approximately\s+([\d.,]+)\s+ada to proceed/
   );
 
+  const [, adaToProceed] = adaToProceedRegex.exec(error.message) ?? [];
```

Comment:
The `adaToProceed` variable shadows the function argument. Actually, what is the intention here? Should the `adaToProceed` from the error message take priority over the passed `adaToProceed`?

Reply:
There is no adaToProceed argument. Could it be you are talking about the return type of the function?

Reply:
Ahh, sorry. You are right. It was my misunderstanding. Everything is good ;)